### ISSUE TITLE
Templates block updates

### ISSUE THREAD

**Issue by Botched1** [BassT23/Proxmox#45]
If you have templates on your hosts the update will stop when it gets past the host update section with errors similar to this:

```
[Info] Starting 901 

you can't start a CT if it's a template
Error during Update --- Exit Code: 255

There are an error during update          
Please check Output here or Logfile
```

**Comment by BassT23** [comment]
THX for testing, and input,
will have a look on this ;)

**Comment by Uruknara** [comment]
The simplest thing to do, in the new update you can skip containers by id and block all the templates one and maybe after will add this so we can detect if it's a template or not and choice to skip it

**Comment by BassT23** [comment]
https://forum.proxmox.com/threads/task-error-you-cant-start-a-ct-if-its-a-template.27749/

could this be helpful?
Because Uruk and I had no problems with our LXC templates.

Otherwise, please give us more input about your problematic template.

Like Uruk said, please test latest beta and exclude your problematic container in the config.

**Comment by Botched1** [comment]
Installed the beta and tried it.

It errors out at both CT and VM templates. So I added those IDs to EXCLUDE.

Next it stopped at a Free BSD based VM I had (no qemu agent). Added that to EXCLUDE.

Then it tried to update an Ubuntu desktop (good!), but did not finish successfully. Maybe the timeout is too low? I didn't see a way to  change that setting though in the config file:
```
[Info] Starting VM 102 
[Info] Waiting for VM 102 to start
This will take some time, ... 30 secounds is set!
[Info] Updating VM 102 : UD-test

--- APT UPDATE ---
"Hit:1 http://us.archive.ubuntu.com/ubuntu jammy InRelease\nHit:2 http://us.archive.ubuntu.com/ubuntu jammy-updates InRelease\nHit:3 http://us.archive.ubuntu.com/ubuntu jammy-backports InRelease\nHit:4 http://security.ubuntu.com/ubuntu jammy-security InRelease\nReading package lists...\n"

--- APT UPGRADE ---
timeout reached, returning pid

--- APT CLEANING ---


[Info] Shutting down VM 102 
```

**Comment by Botched1** [comment]
Additionally, on that 102 VM, if I manually loging, go to terminal and do an aput update, it says there are 175 updates.... So it indeed looks like the updater did not actually update that VM...

![image](https://user-images.githubusercontent.com/19985320/220197515-9bd6b8ee-de14-4313-a991-7955f6d59090.png)

**Comment by BassT23** [comment]
Ok, the problem is; on arch it should do packman update, no apt, ...

What is the output of:
'qm guest cmd 102 get-osinfo | grep name'

**Comment by Botched1** [comment]
It is Ubuntu Desktop...

```
jason@pmc:~# qm guest cmd 102 get-osinfo | grep name
   "name" : "Ubuntu",
   "pretty-name" : "Ubuntu 22.04.2 LTS",
```

**Comment by BassT23** [comment]
Password is not needed, because the script should run as root.
The timeaut is normally set to 30 seconds. I encreased it to 60 seconds now.

please make a 
`bash <(curl -s https://raw.githubusercontent.com/BassT23/Proxmox/beta/install.sh) update`
and try the updater again

**Comment by Botched1** [comment]
Sorry,  same issue.

Still 175 packages that can be updated.

```
             Script is UpToDate
                Version: 3.5.5
```

```
[Info] Updating VM 102 : UD-test

--- APT UPDATE ---
"Hit:1 http://us.archive.ubuntu.com/ubuntu jammy InRelease\nGet:2 http://us.archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]\nGet:3 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]\nGet:4 http://us.archive.ubuntu.com/ubuntu jammy-backports InRelease [107 kB]\nGet:5 http://us.archive.ubuntu.com/ubuntu jammy-updates/main amd64 DEP-11 Metadata [101 kB]\nGet:6 http://us.archive.ubuntu.com/ubuntu jammy-updates/universe amd64 DEP-11 Metadata [267 kB]\nGet:7 http://us.archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 DEP-11 Metadata [940 B]\nGet:8 http://us.archive.ubuntu.com/ubuntu jammy-backports/main amd64 DEP-11 Metadata [7,980 B]\nGet:9 http://us.archive.ubuntu.com/ubuntu jammy-backports/universe amd64 DEP-11 Metadata [12.5 kB]\nFetched 725 kB in 1s (816 kB/s)\nReading package lists...\n"

--- APT UPGRADE ---


--- APT CLEANING ---



[Info] Skipped VM 902 by user


Finished, All Containers Updated.

All fine, have a nice day
```

**Comment by BassT23** [comment]
ok, 175 is extreamly heavy, so the timeout should be "extramly" high, ...
I changed the timeout to 10min

try it out. don't wonder, I don't change the version number.
But you can see it in code - line 302

**Comment by Botched1** [comment]
Same result.

Note that it only stays on  "--- APT UPDATE ---" for about 4 seconds before moving on.

```
[Info] Updating VM 102 : UD-test

--- APT UPDATE ---
"Hit:1 http://us.archive.ubuntu.com/ubuntu jammy InRelease\nGet:2 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]\nGet:3 http://us.archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]\nGet:4 http://us.archive.ubuntu.com/ubuntu jammy-backports InRelease [107 kB]\nFetched 336 kB in 1s (354 kB/s)\nReading package lists...\n"

--- APT UPGRADE ---


--- APT CLEANING ---
```

**Comment by BassT23** [comment]
Please try to make the command from script your own. Whats the output?
`qm guest exec 102 -- bash -c "apt-get -o APT::Get::Always-Include-Phased-Updates=true upgrade -y"`

**Comment by Botched1** [comment]
```
root@pmc:~# qm guest exec 102 -- bash -c "apt-get -o APT::Get::Always-Include-Phased-Updates=true upgrade -y"{
   "err-data" : "E: dpkg was interrupted, you must manually run 'dpkg --configure -a' to correct the problem. \n",
   "exitcode" : 100,
   "exited" : 1
}
root@pmc:~# 
```

**Comment by Botched1** [comment]
Ok, apt was messed up on that ubuntu desktop. 

I couldn't tell from the upgrade script of course as it never really showed that it errored out (although maybe that should be an enhancement request?).

I had to go into the VM and:
```
sudo apt --fix-broken install
```

After that I re-ran update, and it worked as expected.

I also tried the script on a few other ubuntu servers, and they updated with no issues. 

Sorry for the trouble!
